include/oneapi/dpl/internal/random_impl/weibull_distribution.h

```diff
@@ -0,0 +1,202 @@
+// -*- C++ -*-
+//===-- weibull_distribution.h ------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Public header file provides implementation for Weibull Distribution
+
+#ifndef _ONEDPL_WEIBULL_DISTRIBUTION
+#define _ONEDPL_WEIBULL_DISTRIBUTION
+
+namespace oneapi
+{
+namespace dpl
+{
+template <class _RealType = double>
+class weibull_distribution
+{
+  public:
+    // Distribution types
+    using result_type = _RealType;
+    using scalar_type = internal::element_type_t<_RealType>;
+
+    struct param_type
+    {
+        param_type() : param_type(scalar_type{1}) {}
+        param_type(scalar_type __a, scalar_type __b = scalar_type{1}) : a(__a), b(__b) {}
+        scalar_type a;
+        scalar_type b;
+    };
+
+    // Constructors
+    weibull_distribution() : weibull_distribution(scalar_type{1}) {}
+    explicit weibull_distribution(scalar_type __a, scalar_type __b = scalar_type{1}) : a_(__a), b_(__b) {}

```

Comment:
do we need additional checks for parameters?

Reply:
Yes, we do. But we can't throw an execution on the device, so we don't check that.